osf/management/commands/migrate_notifications.py

```diff
@@ -0,0 +1,241 @@
+import logging
+import time
+import signal
+from contextlib import contextmanager
+from django.contrib.contenttypes.models import ContentType
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+from osf.models import NotificationType, NotificationSubscription
+from osf.models.notifications import NotificationSubscriptionLegacy
+from osf.management.commands.populate_notification_types import populate_notification_types
+
+logger = logging.getLogger(__name__)
+
+FREQ_MAP = {
+    'none': 'none',
+    'email_digest': 'weekly',
+    'email_transactional': 'instantly',
+}
+EVENT_NAME_TO_NOTIFICATION_TYPE = {
+    # Provider notifications
+    'new_pending_withdraw_requests': NotificationType.Type.PROVIDER_NEW_PENDING_WITHDRAW_REQUESTS,
+    'contributor_added_preprint': NotificationType.Type.PROVIDER_CONTRIBUTOR_ADDED_PREPRINT,
+    'new_pending_submissions': NotificationType.Type.PROVIDER_NEW_PENDING_SUBMISSIONS,
+    'moderator_added': NotificationType.Type.PROVIDER_MODERATOR_ADDED,
+    'reviews_submission_confirmation': NotificationType.Type.PROVIDER_REVIEWS_SUBMISSION_CONFIRMATION,
+    'reviews_resubmission_confirmation': NotificationType.Type.PROVIDER_REVIEWS_RESUBMISSION_CONFIRMATION,
+    'confirm_email_moderation': NotificationType.Type.PROVIDER_CONFIRM_EMAIL_MODERATION,
+
+    # Node notifications
+    'file_updated': NotificationType.Type.NODE_FILE_UPDATED,
+
+    # Collection submissions
+    'collection_submission_submitted': NotificationType.Type.COLLECTION_SUBMISSION_SUBMITTED,
+    'collection_submission_accepted': NotificationType.Type.COLLECTION_SUBMISSION_ACCEPTED,
+    'collection_submission_rejected': NotificationType.Type.COLLECTION_SUBMISSION_REJECTED,
+    'collection_submission_removed_admin': NotificationType.Type.COLLECTION_SUBMISSION_REMOVED_ADMIN,
+    'collection_submission_removed_moderator': NotificationType.Type.COLLECTION_SUBMISSION_REMOVED_MODERATOR,
+    'collection_submission_removed_private': NotificationType.Type.COLLECTION_SUBMISSION_REMOVED_PRIVATE,
+    'collection_submission_cancel': NotificationType.Type.COLLECTION_SUBMISSION_CANCEL,
+}
+
+
+TIMEOUT_SECONDS = 3600  # 60 minutes timeout
+BATCH_SIZE = 1000  # batch size, can be changed
+
+
+@contextmanager
+def time_limit(seconds):
+    def signal_handler(signum, frame):
+        raise TimeoutError('Migration timed out')
+
+    signal.signal(signal.SIGALRM, signal_handler)
+    signal.alarm(seconds)
+    try:
+        yield
+    finally:
+        signal.alarm(0)
+
+def migrate_legacy_notification_subscriptions(
```

Comment:
@Johnetordoff only migrate users that have changed their defaults, users who only have default settings can be handled in bulk more quickly.